File: 169-Majority-Element/solution.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int size = nums.size();
        int majorityEle = nums.at(0);
        int count = 1;
        
        for(int i = 1; i < size; i++) {
            if(count) {
                if(majorityEle == nums.at(i)) {
                    count++;
                } else {
                    count--;
                }
            } else {
                majorityEle = nums.at(i);
                count++;
            }
        }
        
        return majorityEle;
    }
};

File: 169-Majority-Element/solution.java
class Solution {
    public int majorityElement(int[] nums) {
        int candidate = 0, count = 0;
        for (int n : nums) {
            if (count == 0) {
                candidate = n;
                count++;
            } else {
                if (n == candidate) {
                    count++;
                } else {
                    count--;
                }
            }
        }

        return candidate;
    }
}